projects/soft_999_akbar_pardawalla_fw_103/src/main.c

```diff
@@ -0,0 +1,65 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+void adc_callback(AdcChannel addr, void *context);
+
+void bttn_callback(const GpioAddress *address, void *context) {
+  AdcChannel *channel = context;
+  adc_register_callback(*channel, adc_callback, address);
+}
+
+void adc_callback(AdcChannel addr, void *context) {
```

Comment:
Since these are private functions, they should be static and prefixed with `prv_`. (Also, `btn` is the standard abbreviation for `button`.)